tests/integration/index/query_with_composite_inxed_on_relation_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+// This test was added during https://github.com/sourcenetwork/defradb/issues/2924
+// The issue was that [multiScanNode] that keeps track of how many calls to [Next()] should
+// be made, would call [Init()] and [Start()] every time without tracking, which would result
+// in child nodes being initialized and started multiple times, which in turn created
+// index iterators without closing them.
```

Comment:
praise: Thanks for taking the time to link this to the ticket, it highlights the importance of the test very nicely.